src/main/java/com/minecolonies/coremod/colony/crafting/CustomRecipeManager.java

```diff
@@ -0,0 +1,83 @@
+package com.minecolonies.coremod.colony.crafting;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import com.google.gson.JsonObject;
+
+import static com.minecolonies.coremod.colony.crafting.CustomRecipe.*;
+
+/**
+ * Manager class for tracking Custom recipes during load and use
+ * This class is a singleton
+ */
+public class CustomRecipeManager
+{
+    private static CustomRecipeManager instance = new CustomRecipeManager();
+    private HashMap<String, HashMap<String, CustomRecipe>> recipeMap = new HashMap<>();
+
+    private CustomRecipeManager()
+    {
+    }
+        
+    /**
+     * Get the singleton instance of this class
+     * @return
+     */
+    public static CustomRecipeManager getInstance()
+    {
+        return instance;
+    }
+
+    /**
+     * Add a recipe Json to the manager
+     * @param recipeJson
+     * @param namespace
+     * @param path
+     */
+    public void addRecipe(JsonObject recipeJson, String namespace, String path)
```

Comment:
@NotNull final

Reply:
Since I'm not sure what namespace here means, it might need a better explanation in the javadoc

Reply:
If it's like a ResourceLocation, why don't we use a ResourceLocation here?

Reply:
I'm investigating

Reply:
Ok.Switching.